modules/core/lib/utils/network_manager/network_change_manager.dart

```diff
@@ -0,0 +1,61 @@
+import 'dart:async';
+
+import 'package:connectivity_plus/connectivity_plus.dart';
+
+typedef NetworkCallBack = void Function(NetworkResult result);
+
+abstract class INetworkChangeManager {
+  Future<NetworkResult> checkNetworkOneTime();
+  void handleNetworkChange(NetworkCallBack onChange);
+  void dispose();
+}
+
+class NetworkChangeManager extends INetworkChangeManager {
+  late final Connectivity _connectivity;
+  StreamSubscription<ConnectivityResult>? _subscription;
+  static NetworkChangeManager? _instance;
+  static NetworkChangeManager get instance =>
+      _instance ??= NetworkChangeManager._();
+
+  NetworkChangeManager._() {
+    _connectivity = Connectivity();
+  }
```

Comment:
Instead of creating a singleton structure in this way, let's initialize it as a 'singleton' using the dependency injection system when using it in the application layer.

Reply:
I didn't know how to approach here. Since these files are in the core and core don't know injectable, I didn't want to do it directly and I did it like this. If you want, I can add injectable to this layer, or I can get it from other modules.